core/src/main/java/io/questdb/std/Files.java

```diff
@@ -124,49 +108,44 @@ public static int close(int fd) {
         return -1;
     }
 
-    public static int detach(int fd) {
-        // do not close `stdin` and `stdout`
-        if (fd > 1) {
-            assert auditClose(fd);
-            OPEN_FILE_COUNT.decrementAndGet();
-        }
-        return fd;
-    }
-
-    public static native int copy(long from, long to);
-
     public static int copy(LPSZ from, LPSZ to) {
         return copy(from.ptr(), to.ptr());
     }
 
-    public static native long copyData(int srcFd, int destFd, long offsetSrc, long length);
+    public static long copyData(long srcFd, long destFd, long offsetSrc, long length) {
+        return copyData(toOsFd(srcFd), toOsFd(destFd), offsetSrc, length);
+    }
 
-    public static native long copyDataToOffset(int srcFd, int destFd, long offsetSrc, long offsetDest, long length);
+    public static long copyDataToOffset(long srcFd, long destFd, long offsetSrc, long offsetDest, long length) {
+        return copyDataToOffset(toOsFd(srcFd), toOsFd(destFd), offsetSrc, offsetDest, length);
+    }
 
-    /**
-     * close(fd) should be used instead of this method in most cases
-     * unless you don't need close() sys call to happen.
-     *
-     * @param fd file descriptor
-     */
-    public static void decrementFileCount(int fd) {
-        assert auditClose(fd);
-        OPEN_FILE_COUNT.decrementAndGet();
+    public static int detach(long fd) {
+        int osFd = toOsFd(fd);
+        // do not detach `stdin` and `stdout`
+        if (osFd > 1) {
+            auditClose(fd);
+            OPEN_FILE_COUNT.decrementAndGet();
+            return osFd;
+        }
+        return -1;
     }
 
-    public static native boolean exists(int fd);
+    public static boolean exists(long fd) {
+        return exists(toOsFd(fd));
+    }
 
     public static boolean exists(LPSZ lpsz) {
         return lpsz != null && exists0(lpsz.ptr());
     }
 
-    public static void fadvise(int fd, long offset, long len, int advise) {
+    public static void fadvise(long fd, long offset, long len, int advise) {
         if (Os.isLinux()) {
-            fadvise0(fd, offset, len, advise);
+            fadvise0(toOsFd(fd), offset, len, advise);
         }
     }
 
-    public static native void fadvise0(int fd, long offset, long len, int advise);
+    public static native void fadvise0(long fd, long offset, long len, int advise);
```

Comment:
Seems that `fadvise0` could be a private method.

I haven't checked why it takes a `long fd` now, but its only usage is from the call site that passes the result of `int toOsFd()`.

Reply:
good catch, it's a type, fixed to int in 5f22804767ebade94ec0efa41bd2ce172cc78160